internal/controllers/extension_controller.go

```diff
@@ -505,3 +503,14 @@ func (r *ExtensionReconciler) resolve(ctx context.Context, extension ocv1alpha1.
 	})
 	return resultSet[0], nil
 }
+
+func bundleMetadataFor(bundle *catalogmetadata.Bundle) *ocv1alpha1.BundleMetadata {
+	ver, err := bundle.Version()
```

Comment:
Nit: though bundle cannot be nil after resolution (we must already be having a check for it before), can a check also be added here to avoid nil pointer errors.

Reply:
I was thinking about this as well. If we pass a `nil` pointer into this function, what is our expectation?
1. We could panic on the basis that we always expect to pass a non-nil pointer and a panic is an indication of a programmer error. I could see making that more explicit by checking for nil and panicking with a custom message.
2.  We could return an error from this function, but then we have to consider what we'd want the caller to do with it. It doesn't seem all that great to just return that error up the stack.
3. We could return a `nil` BundleMetadata, which seems fairly reasonable. At that point, we could call it essentially everywhere that we set `InstalledBundle` or `ResolvedBundle`.

I'm between 1 and 3. WDYT?

Reply:
I'm leaning towards 3 personally. I think it makes logical sense that if a `nil` value is provided as an input to this function it returns a `nil` response because if there is no bundle there can be no bundle metadata.

In addition to whichever approach is taken, could we also add a comment to document the expected behavior of this function?

Reply:
My concern (re: option 3) would be side-effects if `nil` is not handled properly in all callers of this function.  If we document that this produces nil for nil input, then it behooves the caller to handle properly.  If undocumented, I think this is this function's responsibility to impose sanity (so option 1).

Reply:
Leaning towards option (1) too, given this is going to be internal, and could only be a programmer's error when they expect a bundle to be not nil, and it ends up being so - as in the previous instances we are explicitly setting the installedResource to be nil. 

To maintain consistency, we could follow (1) do a check and panic if its nil, or follow (3) not explicitly set nil everywhere else, instead call this function.

Reply:
I'll go with option (1):
- Add an explicit panic, so that it is clear to our future selves what the intent is
- Add a comment on the function signature that describes the expectations/behavior.